rllib/utils/tests/test_check_env.py

```diff
@@ -145,12 +147,20 @@ def bad_contains_function(self, x):
                 ValueError, match="The observation collected from "
                 "env"):
             check_env(env)
+        del env
+
+    def test_check_incorrect_space_contains_functions_error(self):
+        def bad_contains_function(self, x):
+            raise ValueError("This is a bad contains function")
+
+        env = make_multi_agent("CartPole-v1")({"num_agents": 2})
         env.observation_space_contains = bad_contains_function
         with pytest.raises(
                 ValueError,
                 match="Your observation_space_contains "
                 "function has some"):
             check_env(env)
+        del env
```

Comment:
I was wondering, why do we need those? :)
Some test case magic?

Reply:
yeah, why do we have to del the local env explicitly?

Reply:
no magic here. Just habit, for preventing memory leaks in tests, a habit I formed when developing [Metaworld](https://github.com/rlworkgroup/metaworld)